src/objects.h

```diff
@@ -38,8 +39,10 @@
 **  'IS_INTOBJ' returns 1 if the object <o> is an (immediate) integer object,
 **  and 0 otherwise.
 */
-#define IS_INTOBJ(o) \
-    ((Int)(o) & 0x01)
+static inline Int IS_INTOBJ(Obj o)
+{
+    return ((Int)o & 0x03) == 0x01;
+}
```

Comment:
This actually changes the behaviour of `IS_INTOBJ`. I think it's the right thing to do, but I feel it's wrong to "sneak" it in like that -- meaning that commit message really should mention it (so that future GAP developers, when tracing this change back through commit history, don't have to wonder whether it was accidental or on purpose).